extensions/sql-bindings/src/test/testContext.ts

```diff
@@ -0,0 +1,382 @@
+	public importBacpac(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public extractDacpac(_: string, __: string, ___: string, ____: string, _____: string, ______: vscodeMssql.TaskExecutionMode): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public createProjectFromDatabase(_: string, __: string, ___: string, ____: string, _____: string, ______: vscodeMssql.ExtractTarget, _______: vscodeMssql.TaskExecutionMode): Thenable<vscodeMssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public deployDacpac(_: string, __: string, ___: boolean, ____: string, _____: vscodeMssql.TaskExecutionMode, ______?: Record<string, string>): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public generateDeployScript(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode, ______?: Record<string, string>): Thenable<mssql.DacFxResult> { return Promise.resolve(mockDacFxResult); }
+	public generateDeployPlan(_: string, __: string, ___: string, ____: vscodeMssql.TaskExecutionMode): Thenable<vscodeMssql.GenerateDeployPlanResult> { return Promise.resolve(mockDacFxResult); }
+	public getOptionsFromProfile(_: string): Thenable<vscodeMssql.DacFxOptionsResult> { return Promise.resolve(mockDacFxMssqlOptionResult); }
+	public validateStreamingJob(_: string, __: string): Thenable<mssql.ValidateStreamingJobResult> { return Promise.resolve(mockDacFxResult); }
+}
+
+export class MockSchemaCompareService implements vscodeMssql.ISchemaCompareService {
+	schemaCompareGetDefaultOptions(): Thenable<vscodeMssql.SchemaCompareOptionsResult> {
+		throw new Error('Method not implemented.');
+	}
+}
+
+export class MockVscodeMssqlIExtension implements vscodeMssql.IExtension {
+	sqlToolsServicePath: string = '';
+	dacFx: vscodeMssql.IDacFxService;
+	schemaCompare: vscodeMssql.ISchemaCompareService;
+	azureFunctions: vscodeMssql.IAzureFunctionsService;
+
+	constructor() {
+		this.dacFx = new MockDacFxMssqlService;
+		this.schemaCompare = new MockSchemaCompareService;
+		this.azureFunctions = new MockAzureFunctionService;
+	}
+	promptForConnection(_?: boolean): Promise<vscodeMssql.IConnectionInfo | undefined> {
+		throw new Error('Method not implemented.');
+	}
+	connect(_: vscodeMssql.IConnectionInfo, __?: boolean): Promise<string> {
+		throw new Error('Method not implemented.');
+	}
+	listDatabases(_: string): Promise<string[]> {
+		throw new Error('Method not implemented.');
+	}
+	getDatabaseNameFromTreeNode(_: vscodeMssql.ITreeNodeInfo): string {
+		throw new Error('Method not implemented.');
+	}
+	getConnectionString(_: string | vscodeMssql.ConnectionDetails, ___?: boolean, _____?: boolean): Promise<string> {
+		throw new Error('Method not implemented.');
+	}
+}
+
+export function createContext(): TestContext {
```

Comment:
This doesn't seem to be used - so you hopefully can get rid of this + a bunch of other stuff in this file.

(I'd also suggest renaming it to something more generic like `mocks` or `testUtils`

Reply:
This is used in the addSQLBindingQuickPick.test.ts in beforeEach and then used to set up the various cases.